clause.go

```diff
@@ -89,30 +89,17 @@ func (c *Clause) PreAction(action Action) *Clause {
 
 // HintAction registers a HintAction (function) for the flag to provide completions
 func (c *Clause) HintAction(action HintAction) *Clause {
-	c.addHintAction(action)
+	c.addUserCompletionWords(action()...)
```

Comment:
I guess I'm confused then about what additional context would be available to the HintAction later?  Since HintAction doesn't take any arguments...